### ISSUE TITLE
README missing dependencies on libdevmapper & btrfs

### ISSUE THREAD

**Issue by jawnsy** [kubernetes-incubator/cri-o#298]
Ran into the following build problems:

```
go build --tags "selinux seccomp btrfs_noversion libdm_no_deferred_remove" -o ocid ./cmd/server/
# github.com/containers/storage/pkg/devicemapper
vendor/src/github.com/containers/storage/pkg/devicemapper/devmapper_wrapper.go:7:26: fatal error: libdevmapper.h: No such file or directory
 #include <libdevmapper.h>
                          ^
compilation terminated.
# github.com/containers/storage/drivers/btrfs
vendor/src/github.com/containers/storage/drivers/btrfs/btrfs.go:8:25: fatal error: btrfs/ioctl.h: No such file or directory
 #include <btrfs/ioctl.h>
                         ^
compilation terminated.
Makefile:55: recipe for target 'ocid' failed
make: *** [ocid] Error 2
```

Needed to install `device-mapper-devel` and `btrfs-progs-devel` (This is your periodic reminder that it's a pain to deal with dependencies for C libraries)

**Comment by surajssd** [comment]
yes faced similar issue when building

```bash
✔ ~/go/src/github.com/kubernetes-incubator/cri-o [master L|…11]                                                                                                                      [338/397]
15:33 $ make
go build --tags "selinux seccomp btrfs_noversion libdm_no_deferred_remove" -o ocid ./cmd/server/
# github.com/containers/image/image
vendor/src/github.com/containers/image/image/image.go:19: validHex redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:18
vendor/src/github.com/containers/image/image/image.go:129: fsLayersSchema1 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:21
vendor/src/github.com/containers/image/image/image.go:136: manifestSchema1 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:42
vendor/src/github.com/containers/image/image/image.go:186: manifestSchema2 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema2.go:37
vendor/src/github.com/containers/image/image/image.go:192: descriptor redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema2.go:30
vendor/src/github.com/containers/image/image/image.go:315: fixManifestLayers redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:186
vendor/src/github.com/containers/image/image/image.go:360: validateV1ID redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:231
vendor/src/github.com/containers/image/image/manifest.go:14: config redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:104
vendor/src/github.com/containers/image/image/manifest.go:19: v1Image redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:108
vendor/src/github.com/containers/image/image/manifest.go:59: genericManifest redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:122
vendor/src/github.com/containers/image/image/manifest.go:59: too many errors
# github.com/containers/storage/pkg/devicemapper
vendor/src/github.com/containers/storage/pkg/devicemapper/devmapper_wrapper.go:7:26: fatal error: libdevmapper.h: No such file or directory
 #include <libdevmapper.h>
                          ^
compilation terminated.
# github.com/containers/storage/drivers/btrfs
vendor/src/github.com/containers/storage/drivers/btrfs/btrfs.go:8:25: fatal error: btrfs/ioctl.h: No such file or directory
 #include <btrfs/ioctl.h>
                         ^
compilation terminated.
Makefile:55: recipe for target 'ocid' failed
make: *** [ocid] Error 2
```

When installed package `device-mapper-devel` then got rid of
```
vendor/src/github.com/containers/storage/pkg/devicemapper/devmapper_wrapper.go:7:26: fatal error: libdevmapper.h: No such file or directory
```

When installed package `btrfs-progs-devel` then got rid of
```
vendor/src/github.com/containers/storage/drivers/btrfs/btrfs.go:8:25: fatal error: btrfs/ioctl.h: No such file or directory
```

But still problem persists

```bash
✘-2 ~/go/src/github.com/kubernetes-incubator/cri-o [master L|…11]
16:34 $ make
go build --tags "selinux seccomp  " -o ocid ./cmd/server/
# github.com/containers/image/image
vendor/src/github.com/containers/image/image/image.go:19: validHex redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:18
vendor/src/github.com/containers/image/image/image.go:129: fsLayersSchema1 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:21
vendor/src/github.com/containers/image/image/image.go:136: manifestSchema1 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:42
vendor/src/github.com/containers/image/image/image.go:186: manifestSchema2 redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema2.go:37
vendor/src/github.com/containers/image/image/image.go:192: descriptor redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema2.go:30
vendor/src/github.com/containers/image/image/image.go:315: fixManifestLayers redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:186
vendor/src/github.com/containers/image/image/image.go:360: validateV1ID redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/docker_schema1.go:231
vendor/src/github.com/containers/image/image/manifest.go:14: config redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:104
vendor/src/github.com/containers/image/image/manifest.go:19: v1Image redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:108
vendor/src/github.com/containers/image/image/manifest.go:59: genericManifest redeclared in this block
        previous declaration at vendor/src/github.com/containers/image/image/image.go:122
vendor/src/github.com/containers/image/image/manifest.go:59: too many errors
Makefile:55: recipe for target 'ocid' failed
make: *** [ocid] Error 2
```

**Comment by surajssd** [comment]
Now all above errors go away, not sure what fixed it!
While testing https://github.com/kubernetes-incubator/cri-o/pull/189 installed `gpgme-devel` and `libassuan-devel` and now master builds, without problems.

**Comment by runcom** [comment]
@nalind could you update the README in #189 to add gpgme, libassuan, btrfs-progs and the like to the installation prereqs? Also, please add a "Close #298" to the first comment of #189 so this gets closed when we merge #189.

**Comment by nalind** [comment]
@runcom will do.
